nodeJS/express/controllers.md

```diff
@@ -108,75 +108,99 @@ function myMiddleware(req, res, next) {
 
-A controller comes into play whenever a request hits the server and a route matches the requested HTTP verb and path. The route determines which controller should handle the request based on the defined middleware chain. The appropriate controller then takes over and performs the necessary actions to fulfill the request. This could involve retrieving data from the model, processing the data, making decisions based on business logic, or updating the model with new data.
+A controller comes into play whenever a request hits the server and a route matches the requested HTTP verb and path. The route determines which controller should handle the request based on the defined middleware chain. The appropriate controller then takes over, performing the necessary actions to fulfill the request.. This could involve retrieving data from the model, processing the data, making decisions based on business logic, or updating the model with new data.
 
-Once the controller has completed its tasks, it passes the processed data to the view, which renders the data into a format suitable for sending back to the client. Typically, this would be HTML. Later, when we cover building APIs, we can also send JSON responses like with the APIs that you've previously encountered e.g. Giphy API.
+Once the controller completes its tasks, it passes the processed data to the view. The view then renders this data into a format suitable for sending back to the client. Typically, this would be HTML. Later, when we cover building APIs, we can also send JSON responses like with the APIs that we've previously used e.g. Giphy API.
 
 The naming conventions for these controllers are usually based on the route they will be attached to e.g. `GET` route -> `getSomething`, `POST` route -> `createSomething`, `DELETE` route -> `deleteSomething`, etc. Nonetheless, there is no fixed rule since Express is not opinionated. It will always be based on you or someone else's conventions, and the requirements of the function.
 
-Let's define our first controller! Create a folder called `controllers` at the root of your Express project. Then create a `userController.js` file within it:
+You should already have your application setup in-place from the previous lesson. Previously, we've defined a route for `/authors/:authorId` to get the author by their ID, let's now define a controller for that route. But to make the example produce something, let's first create a mock database with a function that we can use to retrieve an author by their ID.
```

Comment:
Just a heads up, the current Routes lesson doesn't actually tell people to write the library example code - that was just an example of how routers would be written and why they'd be used.

What I'll do is revisit the Routes lesson and touch up that Routers section to actually make people add that stuff to the basic Express app, then this part's language will make a lot more sense.

I'll get this and the Routes lesson touch-up merged together to avoid confusion over discrepancies in lesson language.

Reply:
Sure I'll wait for that for now, thanks @MaoShizhong

Reply:
@fcasibu Sorry it's taken so long!

Have a look at #28763 (feel free to comment on that one if you have any thoughts!) and if it looks good, you can reference that stuff here as appropriate. That should help improve the flow between lessons.

Reply:
No worries, it looks good. I'll probably work on this on friday when I get my free time. Thanks for your work!